test/e2e/adminapi_kubernetesobjects.go

```diff
@@ -0,0 +1,350 @@
+		var obj unstructured.Unstructured
+		resp, err := adminRequest(ctx, resourceID, http.MethodGet, "/kubernetesobjects", params, nil, &obj)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(resp.StatusCode).To(Equal(http.StatusOK))
+
+		By("comparing it to the actual object retrived via Kubernetes API")
+		cm, err := clients.Kubernetes.CoreV1().ConfigMaps(namespace).Get(objName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(obj.GetNamespace()).To(Equal(cm.Namespace))
+		Expect(obj.GetName()).To(Equal(cm.Name))
+		Expect(obj.GetAnnotations()).To(Equal(cm.Annotations))
+	}
+
+	testConfigMapListOK := func() {
+		By("requesting a list of objects via RP admin API")
+		params := url.Values{
+			"kind":      []string{"configmap"},
+			"namespace": []string{namespace},
+		}
+		var obj unstructured.Unstructured
+		resp, err := adminRequest(ctx, resourceID, http.MethodGet, "/kubernetesobjects", params, nil, &obj)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(resp.StatusCode).To(Equal(http.StatusOK))
+
+		By("comparing names from the list action response with names retrived via Kubernetes API")
+		objs, err := obj.ToList()
+		Expect(err).NotTo(HaveOccurred())
+		var names []string
+		for _, o := range objs.Items {
+			names = append(names, o.GetName())
+		}
+		Expect(names).To(ContainElement(objName))
+	}
+
+	testConfigMapUpdateOK := func() {
+		By("updating the object via RP admin API")
+		body := fmt.Sprintf(
+			`{
+				"kind": "ConfigMap",
+				"apiVersion": "v1",
+				"metadata": {
+					"name": "%s",
+					"namespace": "%s"
+				},
+				"data": {
+					"key": "value"
+				}
+			}`,
+			objName,
+			namespace,
+		)
+		resp, err := adminRequest(ctx, resourceID, http.MethodPost, "/kubernetesobjects", nil, json.RawMessage(body), nil)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(resp.StatusCode).To(Equal(http.StatusOK))
+
+		By("checking that the object changed via Kubernetes API")
+		cm, err := clients.Kubernetes.CoreV1().ConfigMaps(namespace).Get(objName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(cm.Namespace).To(Equal(namespace))
+		Expect(cm.Name).To(Equal(objName))
+		Expect(cm.Data).To(Equal(map[string]string{"key": "value"}))
+	}
+
+	testConfigMapDeleteOK := func() {
+		By("deleting the object via RP admin API")
+		params := url.Values{
+			"kind":      []string{"configmap"},
+			"namespace": []string{namespace},
+			"name":      []string{objName},
+		}
+		resp, err := adminRequest(ctx, resourceID, http.MethodDelete, "/kubernetesobjects", params, nil, nil)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(resp.StatusCode).To(Equal(http.StatusOK))
+
+		By("waiting for the configmap to be deleted")
+		err = wait.PollImmediate(10*time.Second, time.Minute, func() (bool, error) {
+			_, err = clients.Kubernetes.CoreV1().ConfigMaps(namespace).Get(objName, metav1.GetOptions{})
+			if err == nil {
+				return false, nil
+			}
+			if !errors.IsNotFound(err) {
+				return false, err
+			}
+			return true, nil
+		})
```

Comment:
Poll shouldn't be necessary - either it should not exist or there should be a deletionTimestamp set in the metadata.

Reply:
It prevents flakines: we wait for the object to be completely deleted before we start other tests which may (or may not) rely on the object.